packet/packet.go

```diff
@@ -213,6 +226,53 @@ func (packet *Packet) ParseL3() {
 	packet.L3 = unsafe.Pointer(packet.unparsed())
 }
 
+// Correctly returns EtherType from Ethernet header or VLAN header.
+func (packet *Packet) GetEtherType() uint16 {
+	if packet.Ether.EtherType == SwapBytesUint16(VLANNumber) {
+		vptr := uintptr(unsafe.Pointer(packet.Ether)) + EtherLen
```

Comment:
Here and below: we have
"func (packet *Packet) unparsed() uintptr {
	return uintptr(unsafe.Pointer(packet.Ether)) + EtherLen
}"
for this common expression